evap/staff/tests/test_views.py

```diff
@@ -904,40 +904,69 @@ def test_success_with_data(self):
 
 
-class TestSemesterAssignView(WebTestStaffMode):
+class TestSemesterQuestionnaireAssignment(WebTestStaffMode):
     @classmethod
     def setUpTestData(cls):
         cls.manager = make_manager()
-        cls.semester = baker.make(Semester)
-        cls.url = f"/staff/semester/{cls.semester.pk}/assign"
+        semester = baker.make(Semester)
+        cls.url = f"/staff/semester/{semester.pk}/assign"
 
-        lecture_type = baker.make(CourseType, name_de="Vorlesung", name_en="Lecture")
-        seminar_type = baker.make(CourseType, name_de="Seminar", name_en="Seminar")
-        cls.questionnaire = baker.make(Questionnaire, type=Questionnaire.Type.TOP)
+        cls.responsible = baker.make(UserProfile)
 
-        evaluation1 = baker.make(Evaluation, course__type=seminar_type, course__semester=cls.semester)
-        evaluation2 = baker.make(Evaluation, course__type=lecture_type, course__semester=cls.semester)
+        cls.questionnaires = baker.make(Questionnaire, type=Questionnaire.Type.TOP, _quantity=2)
+        cls.questionnaire_contributor, cls.questionnaire_responsible = baker.make(
+            Questionnaire, type=Questionnaire.Type.CONTRIBUTOR, _quantity=2
+        )
+        cls.course_types = baker.make(CourseType, _quantity=3)
+        cls.evaluations = baker.make(
+            Evaluation,
+            course__semester=semester,
+            course__responsibles=[cls.responsible],
+            course__type=iter(cls.course_types),
+            _quantity=3,
+        )
         baker.make(
             Contribution,
-            contributor=baker.make(UserProfile),
-            evaluation=iter([evaluation1, evaluation2]),
+            contributor=cls.responsible,
+            evaluation=iter(cls.evaluations),
             role=Contribution.Role.EDITOR,
             textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-            _fill_optional=["contributor"],
-            _quantity=2,
-            _bulk_create=True,
+            _quantity=3,
         )
 
-    def test_assign_questionnaires(self):
-        page = self.app.get(self.url, user=self.manager)
-        assign_form = page.forms["questionnaire-assign-form"]
-        assign_form["Seminar"] = [self.questionnaire.pk]
-        assign_form["Lecture"] = [self.questionnaire.pk]
-        page = assign_form.submit().follow()
+    def test_questionnaire_assignment(self):
+        page = self.app.get(self.url, user=self.manager, status=200)
+        form = page.forms["questionnaire-assign-form"]
+        form[f"general-{self.course_types[0].id}"] = [self.questionnaires[0].pk, self.questionnaires[1].pk]
+        form[f"general-{self.course_types[1].id}"] = [self.questionnaires[1].pk]
+        form[f"contributor-{self.course_types[0].id}"] = [self.questionnaire_responsible.pk]
+        form["all-contributors"] = [self.questionnaire_contributor.pk]
+
+        response = form.submit().follow()
+        self.assertIn("Successfully", str(response))
```

Comment:
There is `assertContains` to assert a HttpResponse contains some specific text:
```suggestion
        self.assertContains(response, "Successfully")
```